File: GXpert/GXpert.Web/Modules/Content/ContentCategory/ContentCategoryColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.Content.Columns;

[ColumnsScript("Content.ContentCategory")]
[BasedOnRow(typeof(ContentCategoryRow), CheckNames = true)]
public class ContentCategoryColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Name { get; set; }
    public string Description { get; set; }
    public bool IsGlobal { get; set; }
    public int MediaCount { get; set; }
    public string Thumbnail { get; set; }
    public string ListingThumbnail { get; set; }
    public DateTime InsertDate { get; set; }
    public int InsertUserId { get; set; }
    public DateTime UpdateDate { get; set; }
    public int UpdateUserId { get; set; }
    public short IsActive { get; set; }
}

File: GXpert/GXpert.Web/Modules/ServerTypes/Content/ContentCategoryColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { ContentCategoryRow } from "./ContentCategoryRow";

export interface ContentCategoryColumns {
    Id: Column<ContentCategoryRow>;
    Name: Column<ContentCategoryRow>;
    Description: Column<ContentCategoryRow>;
    IsGlobal: Column<ContentCategoryRow>;
    MediaCount: Column<ContentCategoryRow>;
    Thumbnail: Column<ContentCategoryRow>;
    ListingThumbnail: Column<ContentCategoryRow>;
}

export class ContentCategoryColumns extends ColumnsBase<ContentCategoryRow> {
    static readonly columnsKey = 'Content.ContentCategory';
    static readonly Fields = fieldsProxy<ContentCategoryColumns>();
}